### ISSUE TITLE
On Windows 11 Pro, aspera-cli-4.18.0 is not running properly.

### ISSUE THREAD

**Issue by blogsook** [IBM/aspera-cli#154]
Hello.
As shown below, aspera-cli-4.18.0 does not run properly on Windows 11 Pro.

```cmd
C:> systeminfo

호스트 이름:             HOSTNAME
OS 이름:                 Microsoft Windows 11 Pro
OS 버전:                 10.0.22631 N/A 빌드 22631


C:\Users\username\repo\aspera-cli-local\cache>gem -v
3.5.17

C:\Users\username\repo\aspera-cli-local\cache>gem install aspera-cli
Successfully installed aspera-cli-4.18.0
Parsing documentation for aspera-cli-4.18.0
Installing ri documentation for aspera-cli-4.18.0
Done installing documentation for aspera-cli after 1 seconds
1 gem installed

C:\Users\username\repo\aspera-cli-local\cache>gem info aspera-cli

*** LOCAL GEMS ***

aspera-cli (4.18.0)
    Author: Laurent Martin
    Homepage: https://github.com/IBM/aspera-cli
    License: Apache-2.0
    Installed at: C:/Ruby33-x64/lib/ruby/gems/3.3.0

    Execute actions using command line on IBM Aspera Server products:
    Aspera on Cloud, Faspex, Shares, Node, Console, Orchestrator, High
    Speed Transfer Server

C:\Users\username\repo\aspera-cli-local\cache>ascli -h
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed options: ["-h"]
Use --log-level=debug to get more details.


C:\Users\username\repo\aspera-cli-local\cache>ascli -h --log-level=debug
D, [2024-08-21T11:04:09.130978 #15112] DEBUG -- : env={}
D, [2024-08-21T11:04:09.131199 #15112] DEBUG -- : declare: interactive: accessor
D, [2024-08-21T11:04:09.131283 #15112] DEBUG -- : set attr obj interactive (#<Aspera::Cli::Manager:0x000001db0fac35c8>,ask_missing_mandatory)
D, [2024-08-21T11:04:09.131373 #15112] DEBUG -- : AttrAccessor: interactive: Aspera::Cli::Manager.ask_missing_mandatory: writer=true
D, [2024-08-21T11:04:09.131443 #15112] DEBUG -- : (accessor) get interactive=false
D, [2024-08-21T11:04:09.131821 #15112] DEBUG -- : on_args=["Use interactive input of missing params: \e[1m\e[31mno\e[0m\e[0m, yes", "--interactive=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:04:09.132030 #15112] DEBUG -- : declare: ask_options: accessor
D, [2024-08-21T11:04:09.132105 #15112] DEBUG -- : set attr obj ask_options (#<Aspera::Cli::Manager:0x000001db0fac35c8>,ask_missing_optional)
D, [2024-08-21T11:04:09.132413 #15112] DEBUG -- : AttrAccessor: ask_options: Aspera::Cli::Manager.ask_missing_optional: writer=true
D, [2024-08-21T11:04:09.132520 #15112] DEBUG -- : (accessor) get ask_options=false
D, [2024-08-21T11:04:09.132631 #15112] DEBUG -- : on_args=["Ask even optional options: \e[1m\e[31mno\e[0m\e[0m, yes", "--ask-options=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:04:09.132819 #15112] DEBUG -- : parse_options!
D, [2024-08-21T11:04:09.132890 #15112] DEBUG -- : consume_option_pairs: set
D, [2024-08-21T11:04:09.132933 #15112] DEBUG -- : consume_option_pairs: env
D, [2024-08-21T11:04:09.132974 #15112] DEBUG -- : before parse
D, [2024-08-21T11:04:09.133449 #15112] DEBUG -- : After parse
D, [2024-08-21T11:04:09.133518 #15112] DEBUG -- : remains: []
D, [2024-08-21T11:04:09.133600 #15112] DEBUG -- : add_cmd_line_options:commands/arguments=[],options=["-h", "--log-level=debug"]
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed options: ["-h", "--log-level=debug"]
C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/environment.rb:126:in `terminal_supports_unicode?': undefined method `include?' for nil (NoMethodError)

        @terminal_supports_unicode = terminal? && ENV.values_at('LC_ALL', 'LC_CTYPE', 'LANG').compact.first.include?('UTF-8') if @terminal_supports_unicode.nil?
                                                                                                           ^^^^^^^^^
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/formatter.rb:200:in `declare_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:100:in `init_agents_and_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:248:in `process_command_line'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/bin/ascli:24:in `<top (required)>'
        from C:/Ruby33-x64/bin/ascli:32:in `load'
        from C:/Ruby33-x64/bin/ascli:32:in `<main>'
```

**Comment by blogsook** [comment]
Add.
C:\Users\username>ascli config plugin list
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed values: ["config", "plugin", "list"]
Use --log-level=debug to get more details.

C:\Users\shlee>ascli config plugin list  --log-level=debug
D, [2024-08-21T11:16:09.544695 #7040] DEBUG -- : env={}
D, [2024-08-21T11:16:09.544901 #7040] DEBUG -- : declare: interactive: accessor
D, [2024-08-21T11:16:09.545004 #7040] DEBUG -- : set attr obj interactive (#<Aspera::Cli::Manager:0x0000024e39684898>,ask_missing_mandatory)
D, [2024-08-21T11:16:09.545104 #7040] DEBUG -- : AttrAccessor: interactive: Aspera::Cli::Manager.ask_missing_mandatory: writer=true
D, [2024-08-21T11:16:09.545194 #7040] DEBUG -- : (accessor) get interactive=false
D, [2024-08-21T11:16:09.545363 #7040] DEBUG -- : on_args=["Use interactive input of missing params: \e[1m\e[31mno\e[0m\e[0m, yes", "--interactive=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:16:09.545455 #7040] DEBUG -- : declare: ask_options: accessor
D, [2024-08-21T11:16:09.545544 #7040] DEBUG -- : set attr obj ask_options (#<Aspera::Cli::Manager:0x0000024e39684898>,ask_missing_optional)
D, [2024-08-21T11:16:09.545624 #7040] DEBUG -- : AttrAccessor: ask_options: Aspera::Cli::Manager.ask_missing_optional: writer=true
D, [2024-08-21T11:16:09.545689 #7040] DEBUG -- : (accessor) get ask_options=false
D, [2024-08-21T11:16:09.545783 #7040] DEBUG -- : on_args=["Ask even optional options: \e[1m\e[31mno\e[0m\e[0m, yes", "--ask-options=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:16:09.545857 #7040] DEBUG -- : parse_options!
D, [2024-08-21T11:16:09.545907 #7040] DEBUG -- : consume_option_pairs: set
D, [2024-08-21T11:16:09.545954 #7040] DEBUG -- : consume_option_pairs: env
D, [2024-08-21T11:16:09.546000 #7040] DEBUG -- : before parse
D, [2024-08-21T11:16:09.546068 #7040] DEBUG -- : After parse
D, [2024-08-21T11:16:09.546114 #7040] DEBUG -- : remains: []
D, [2024-08-21T11:16:09.546170 #7040] DEBUG -- : add_cmd_line_options:commands/arguments=["config", "plugin", "list"],options=["--log-level=debug"]
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed options: ["--log-level=debug"]
ERROR: Argument: unprocessed values: ["config", "plugin", "list"]
C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/environment.rb:126:in `terminal_supports_unicode?': undefined method `include?' for nil (NoMethodError)

        @terminal_supports_unicode = terminal? && ENV.values_at('LC_ALL', 'LC_CTYPE', 'LANG').compact.first.include?('UTF-8') if @terminal_supports_unicode.nil?
                                                                                                           ^^^^^^^^^
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/formatter.rb:200:in `declare_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:100:in `init_agents_and_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:248:in `process_command_line'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/bin/ascli:24:in `<top (required)>'
        from C:/Ruby33-x64/bin/ascli:32:in `load'
        from C:/Ruby33-x64/bin/ascli:32:in `<main>'

ascli shares health  --url=https://192.168.XXX.77 --username=test --password=qwer1234 --insecure=yes
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed options: ["--url=https://192.168.XXX.77", "--username=test", "--password=qwer1234", "--insecure=yes"]
ERROR: Argument: unprocessed values: ["shares", "health"]
Use --log-level=debug to get more details.

C:\Users\shlee>ascli shares health  --url=https://192.168.XXX.77 --username=test --password=qwer1234 --insecure=yes --log-level=debug
D, [2024-08-21T11:17:40.757034 #848] DEBUG -- : env={}
D, [2024-08-21T11:17:40.757225 #848] DEBUG -- : declare: interactive: accessor
D, [2024-08-21T11:17:40.757322 #848] DEBUG -- : set attr obj interactive (#<Aspera::Cli::Manager:0x000001f305d51b88>,ask_missing_mandatory)
D, [2024-08-21T11:17:40.757396 #848] DEBUG -- : AttrAccessor: interactive: Aspera::Cli::Manager.ask_missing_mandatory: writer=true
D, [2024-08-21T11:17:40.757500 #848] DEBUG -- : (accessor) get interactive=false
D, [2024-08-21T11:17:40.757668 #848] DEBUG -- : on_args=["Use interactive input of missing params: \e[1m\e[31mno\e[0m\e[0m, yes", "--interactive=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:17:40.757758 #848] DEBUG -- : declare: ask_options: accessor
D, [2024-08-21T11:17:40.757825 #848] DEBUG -- : set attr obj ask_options (#<Aspera::Cli::Manager:0x000001f305d51b88>,ask_missing_optional)
D, [2024-08-21T11:17:40.757910 #848] DEBUG -- : AttrAccessor: ask_options: Aspera::Cli::Manager.ask_missing_optional: writer=true
D, [2024-08-21T11:17:40.757998 #848] DEBUG -- : (accessor) get ask_options=false
D, [2024-08-21T11:17:40.758127 #848] DEBUG -- : on_args=["Ask even optional options: \e[1m\e[31mno\e[0m\e[0m, yes", "--ask-options=ENUM", [:yes, true, :no, false]]
D, [2024-08-21T11:17:40.758211 #848] DEBUG -- : parse_options!
D, [2024-08-21T11:17:40.758272 #848] DEBUG -- : consume_option_pairs: set
D, [2024-08-21T11:17:40.758366 #848] DEBUG -- : consume_option_pairs: env
D, [2024-08-21T11:17:40.758432 #848] DEBUG -- : before parse
D, [2024-08-21T11:17:40.758528 #848] DEBUG -- : After parse
D, [2024-08-21T11:17:40.758587 #848] DEBUG -- : remains: []
D, [2024-08-21T11:17:40.758664 #848] DEBUG -- : add_cmd_line_options:commands/arguments=["shares", "health"],options=["--url=https://192.168.30.77", "--username=test", "--password=🔑
ERROR: Other(NoMethodError): undefined method `include?' for nil
ERROR: Argument: unprocessed options: ["--url=https://192.168.XXX.77", "--username=test", "--password=qwer1234", "--insecure=yes", "--log-level=debug"]
ERROR: Argument: unprocessed values: ["shares", "health"]
C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/environment.rb:126:in `terminal_supports_unicode?': undefined method `include?' for nil (NoMethodError)

        @terminal_supports_unicode = terminal? && ENV.values_at('LC_ALL', 'LC_CTYPE', 'LANG').compact.first.include?('UTF-8') if @terminal_supports_unicode.nil?
                                                                                                           ^^^^^^^^^
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/formatter.rb:200:in `declare_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:100:in `init_agents_and_options'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/lib/aspera/cli/main.rb:248:in `process_command_line'
        from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/aspera-cli-4.18.0/bin/ascli:24:in `<top (required)>'
        from C:/Ruby33-x64/bin/ascli:32:in `load'
        from C:/Ruby33-x64/bin/ascli:32:in `<main>'

**Comment by laurent-martin** [comment]
Hi,
Yes, thanks , it's a known issue already fixed in development branch, which will be released soon.
For the time being use an older version, or the beta from here:

https://ibm.biz/aspera-cli-beta

**Comment by laurent-martin** [comment]
Note: it should be fixed in latest release: 4.18.1
